zeeguu_api/api/dashboard.py

```diff
@@ -0,0 +1,288 @@
+    zeeguu.db.session.add(TeacherCohortMap(user, cohort))
+    zeeguu.db.session.commit()
+    return 'added teacher_class relationship'
+
+#Checks if the inputted invite code is already in use.
+@api.route("/check_invite_code/<invite_code>", methods=["GET"])
+@with_session
+def check_inv_code(invite_code):
+    c = Cohort.query.filter_by(inv_code=invite_code).first()
+    if c is None:
+        return jsonify(1)
+    return jsonify(0)
+
+
+# creates a class in the data base. Requires form input (inv_code, class_name, class_language_id, max_students, teacher_id)
+@api.route("/add_class", methods=["POST"])
+@with_session
+def add_class():
+    inv_code = request.form.get("inv_code")
+    class_name = request.form.get("class_name")
+    class_language_id = request.form.get("class_language_id")
+    if len(class_language_id) != 2:
+        flask.abort(400)
+    class_language = Language.find_or_create(class_language_id)
+    teacher_id = flask.g.user.id
+    max_students = request.form.get("max_students")
+    if int(max_students) < 1:
+        flask.abort(400)
+
+    try:
+        c = Cohort(inv_code, class_name, class_language, max_students)
+        zeeguu.db.session.add(c)
+        zeeguu.db.session.commit()
+        link_teacher_class(teacher_id, c.id)
+        return jsonify(1)
+    except ValueError:
+        #print("value error")
+        flask.abort(400)
+    except sqlalchemy.exc.IntegrityError:
+        #print("integ error")
+        flask.abort(400)
+
+
+# creates user and adds them to a cohort
+@api.route("/add_user_with_class", methods=['POST'])
+def add_user_with_class():
+    email = request.form.get("email")
+    password = request.form.get("password")
+    username = request.form.get("username")
+    inv_code = request.form.get("inv_code")
+
+    cohort_id = Cohort.get_id(inv_code)
+    cohort = Cohort.find(cohort_id)
+
+    if not cohort is None:
+        if not password is None:
+            # Checks to see if there is space, if there is it increments the cur_students variables.
+            # Therefor it is essential that you ensure you add the student if this function returns true.
+            # Hence it being placed after other checks.
+            # However, if an exeption is caught, this error is handled.
+            if cohort.request_join():
+                try:
+                    user = User(email, username, password)
+                    user.cohort = cohort
+                    zeeguu.db.session.add(user)
+                    zeeguu.db.session.commit()
+                    return 'user created'
+                except ValueError:
+                    cohort.undo_join()
+                    flask.abort(400)
+                except sqlalchemy.exc.IntegrityError:
+                    cohort.undo_join()
+                    flask.abort(400)
+            return 'no more space in class!'
+    return 'failed :('
+
+
+#Get user bookmarks
+@api.route("/get_user_stats/<id>", methods=["GET", "POST"])
```

Comment:
why is this named get user stats if it gets bookmarks?

Reply:
Renamed to get_user_bookmarks